rust/src/state_machine/idle.rs

```diff
@@ -0,0 +1,62 @@
+use super::{sum::Sum, CoordinatorState, Request, State, StateMachine};
+use crate::{
+    coordinator::RoundSeed,
+    crypto::{ByteObject, SigningKeySeed},
+};
+use sodiumoxide::crypto::hash::sha256;
+use tokio::sync::mpsc;
+
+#[derive(Debug)]
+pub struct Idle;
+
+impl State<Idle> {
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(
+        coordinator_state: CoordinatorState,
+        request_rx: mpsc::UnboundedReceiver<Request>,
+    ) -> StateMachine {
+        info!("state transition");
+        StateMachine::Idle(Self {
+            _inner: Idle,
+            coordinator_state,
+            request_rx,
+        })
+    }
+
+    pub async fn next(mut self) -> StateMachine {
+        self.clear_round_dicts();
```

Comment:
You are right, it would make more sense to clean it after the sum2 phase. 
The reason I did it this way was because I didn't want to add `clear_round_dicts` again in error state.